### PR TITLE
feat: refactor wishlist modal with formly

### PR DESCRIPTION
**Pull request by MaxKless**
## PR Type<br /><br />
<br /><br />[ ] Bugfix<br />[ ] Feature<br />[ ] Code style update (formatting, local variables)<br />[x] Refactoring (no functional changes, no API changes)<br />[ ] Build-related changes<br />[ ] CI-related changes<br />[ ] Documentation content changes<br />[ ] Application / infrastructure changes<br />[ ] Other: <br /><br />
## What Is the Current Behavior? <br />  <br/>
Currently the wishlist modal uses ish-input and the like to construct its form. We want to refactor it to using formly.<br /><br />
## What Is the New Behavior?<br /> <br/>
Now the wishlist modal uses formly and is built more functionally.<br /> <br/>
## Does this PR Introduce a Breaking Change?<br /><br />
[ ] Yes<br />[x] No<br /><br />## Other Information<br />

[AB#64841](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/64841)

### CODE REVIEW

**Review by iwiederhold — DISMISSED** [review]

**Review by M-Behr — APPROVED** [review]

File: src/app/extensions/wishlists/shared/select-wishlist-modal/select-wishlist-modal.component.ts [hunk 1]
```diff
@@ -27,7 +29,7 @@ import { Wishlist } from '../../models/wishlist/wishlist.model';
 @Component({
   selector: 'ish-select-wishlist-modal',
   templateUrl: './select-wishlist-modal.component.html',
-  changeDetection: ChangeDetectionStrategy.OnPush,
+  changeDetection: ChangeDetectionStrategy.Default,
```
Comment (SGrueber) [inline src/app/extensions/wishlists/shared/select-wishlist-modal/select-wishlist-modal.component.ts:32]: Is this change necessary?

File: src/app/extensions/wishlists/shared/select-wishlist-modal/select-wishlist-modal.component.ts [hunk 2]
```diff
@@ -40,141 +42,181 @@ export class SelectWishlistModalComponent implements OnInit, OnDestroy {
...
-  private formInit() {
-    this.updateWishlistForm = this.fb.group({
-      wishlist: [
-        this.wishlistOptions && this.wishlistOptions.length > 0 ? this.wishlistOptions[0].value : 'newList',
-        Validators.required,
-      ],
-      newWishlist: [this.newWishlistInitValue, Validators.required],
-    });
-  }
+    this.singleFieldConfig = [
```
Comment (SGrueber) [inline src/app/extensions/wishlists/shared/select-wishlist-modal/select-wishlist-modal.component.ts:67]: maybe a comment about the blocks would be helpful. What is meant with singleFieldConfig,  wishlistOptions and multifieldconfig?

File: src/app/extensions/wishlists/shared/select-wishlist-modal/select-wishlist-modal.component.ts [hunk 3]
```diff
@@ -40,141 +42,181 @@ export class SelectWishlistModalComponent implements OnInit, OnDestroy {
...
+      },
+    ];
 
-  private determineSelectOptions() {
-    let currentWishlist: Wishlist;
-    this.wishlistsFacade.currentWishlist$.pipe(take(1), takeUntil(this.destroy$)).subscribe(w => (currentWishlist = w));
-    this.wishlistsFacade.wishlists$.pipe(takeUntil(this.destroy$)).subscribe(wishlists => {
-      if (wishlists && wishlists.length > 0) {
-        this.wishlistOptions = wishlists.map(wishlist => ({
+    this.wishlistOptions$ = this.wishlistsFacade.wishlists$.pipe(
```
Comment (SGrueber) [inline src/app/extensions/wishlists/shared/select-wishlist-modal/select-wishlist-modal.component.ts:83]: can you move this wishlistOptions functionality to the  wishlist.facade? with a boolean parameter like 'forMoving' or something like that

File: src/app/extensions/wishlists/shared/select-wishlist-modal/select-wishlist-modal.component.ts [hunk 4]
```diff
@@ -231,4 +274,9 @@ export class SelectWishlistModalComponent implements OnInit, OnDestroy {
       ? 'account.wishlists.add_to_wishlist.confirmation'
       : 'account.wishlists.move_wishlist_item.confirmation';
   }
+
+  ngOnDestroy() {
+    this.destroy$.next();
+    this.destroy$.complete();
+  }
```
Comment (SGrueber) [inline src/app/extensions/wishlists/shared/select-wishlist-modal/select-wishlist-modal.component.ts:281]: This component has still a lot of lines of code. Consider to outsource the multiplefield form functionality in an extra component

File: src/app/shared/formly/dev/testing/formly-testing.module.ts [hunk 5]
```diff
@@ -22,6 +22,11 @@ class CheckboxFieldComponent extends FieldType {}
 })
 class FieldsetFieldComponent extends FieldType {}
 
+@Component({
+  template: `RadioFielComponent: {{ field.key }} {{ field.type }} {{ to | json }} `,
```
Comment (SGrueber) [inline src/app/shared/formly/dev/testing/formly-testing.module.ts:26]: typo?

**Review by SGrueber — CHANGES_REQUESTED** [review]

File: package-lock.json [hunk 6]
```diff
@@ -5814,6 +5814,16 @@
       "integrity": "sha512-jDctJ/IVQbZoJykoeHbhXpOlNBqGNcwXJKJog42E5HDPUwQTSdjCHdihjj0DlnheQ7blbT6dHOafNAiS8ooQKA==",
       "dev": true
     },
+    "bindings": {
```
[the remark this answers is not in the document]
Comment (MaxKless) [reply package-lock.json:5817]: no, just happened when I npm installed

File: src/app/extensions/wishlists/facades/wishlists.facade.ts [hunk 7]
```diff
@@ -33,6 +34,25 @@ export class WishlistsFacade {
   wishlistLoading$: Observable<boolean> = this.store.pipe(select(getWishlistsLoading));
   wishlistError$: Observable<HttpError> = this.store.pipe(select(getWishlistsError));
 
+  wishlistSelectOptions$(filterCurrent = true) {
```
Comment (SGrueber) [inline src/app/extensions/wishlists/facades/wishlists.facade.ts:37]: nice :-)

File: src/app/extensions/wishlists/shared/select-wishlist-form/select-wishlist-form.component.spec.ts [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  beforeEach(() => {
+    when(wishlistFacade.wishlistSelectOptions$(anything())).thenReturn(
+      of([{ value: 'wishlistId', label: 'wishlistName' }])
+    );
+    fixture = TestBed.createComponent(SelectWishlistFormComponent);
+    component = fixture.componentInstance;
+    element = fixture.nativeElement;
+  });
+
+  it('should be created', () => {
```
Comment (SGrueber) [inline src/app/extensions/wishlists/shared/select-wishlist-form/select-wishlist-form.component.spec.ts:36]: some more tests would be useful, maybe some tests can be moved from the select-wishlist-modal

**Review by SGrueber — CHANGES_REQUESTED** [review]

**Pull request merged**